### PR TITLE
Yuhang/v3.0 preview.1

### PR DESCRIPTION
**Pull request by v-yuhang**
Upgrade the API of module compose with 3.0

### CODE REVIEW

File: src/common/utils.ts [hunk 1]
```diff
@@ -539,7 +539,7 @@ export class URIUtils {
             }
             return retParams;
         }
-        const toPath = compile(pathTemplate, { encode: encodeURIComponent });
+        const toPath = compile(pathTemplate);
```
Comment (buddhawang) [inline src/common/utils.ts:542]: question: why do yo need to update this?

File: src/react/components/pages/modelCompose/modelCompose.tsx [hunk 2]
```diff
@@ -155,16 +155,16 @@ export default class ModelComposePage extends React.Component<IModelComposePageP
...
+            // {
+            //     key: "column4",
+            //     name: strings.modelCompose.column.status.headerName,
+            //     fieldName: strings.modelCompose.column.status.fieldName,
+            //     minWidth: 50,
+            //     maxWidth: 100,
+            //     isResizable: true,
+            //     onColumnClick: this.handleColumnClick,
+            //     onRender: (model: IModel) => <span>{model.status}</span>
+            // },
```
Comment (buddhawang) [inline src/react/components/pages/modelCompose/modelCompose.tsx:167]: remove

File: src/react/components/pages/modelCompose/modelCompose.tsx [hunk 3]
```diff
@@ -177,16 +177,16 @@ export default class ModelComposePage extends React.Component<IModelComposePageP
...
+            // {
+            //     key: "column6",
+            //     name: strings.modelCompose.column.lastUpdated.headerName,
+            //     fieldName: strings.modelCompose.column.lastUpdated.fieldName,
+            //     minWidth: 175,
+            //     maxWidth: 175,
+            //     isResizable: true,
+            //     onColumnClick: this.handleColumnClick,
+            //     onRender: (model: IModel) => <span>{new Date(model.lastUpdatedDateTime).toLocaleString()}</span>,
+            // },
```
Comment (buddhawang) [inline src/react/components/pages/modelCompose/modelCompose.tsx:189]: remove

**Review by buddhawang — CHANGES_REQUESTED** [review]
can you remove changes in predictPage.tsx and trainPage.tsx? I'd like to keep this change primary on model compose.

File: src/react/components/pages/modelCompose/modelCompose.tsx (hunk 2 above)
Comment (v-yuhang) [reply src/react/components/pages/modelCompose/modelCompose.tsx:167]: The previous problem was to solve the problem that the generated URL contains multiple characters: / etc. now it is no longer needed. I will change it back

File: src/react/components/pages/modelCompose/modelCompose.tsx (hunk 3 above)
Comment (v-yuhang) [reply src/react/components/pages/modelCompose/modelCompose.tsx:189]: I will

File: src/common/utils.ts (hunk 1 above)
Comment (v-yuhang) [reply src/common/utils.ts:542]: The previous problem was to solve the problem that the generated URL contains multiple characters: / etc. now it is no longer needed. I will change it back

File: src/react/components/pages/modelCompose/modelCompose.tsx (hunk 2 above)
Comment (v-yuhang) [reply src/react/components/pages/modelCompose/modelCompose.tsx:167]: I will

**Comment by v-yuhang** [comment]
OK, I can finish it
________________________________
发件人: Buddha Wang ***@***.***>
发送时间: 2021年7月26日 8:51
收件人: microsoft/OCR-Form-Tools ***@***.***>
抄送: Yuhang Zhao (Pactera) ***@***.***>; Author ***@***.***>
主题: Re: [microsoft/OCR-Form-Tools] Yuhang/v3.0 preview.1 (#953)


@buddhawang requested changes on this pull request.

can you remove changes in predictPage.tsx and trainPage.tsx? I'd like to keep this change primary on model compose.

________________________________

In src/react/components/pages/modelCompose/modelCompose.tsx<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FOCR-Form-Tools%2Fpull%2F953%23discussion_r676724005&data=04%7C01%7Cv-yuhangzhao%40microsoft.com%7C50eacb621166458e197408d9504d4a0b%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637629115171014003%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=iLeU6vVPwxIN9AR%2F%2F4fcW77y6gzZ%2F0Wlu8PcVAwOUro%3D&reserved=0>:

> +            // {
+            //     key: "column4",
+            //     name: strings.modelCompose.column.status.headerName,
+            //     fieldName: strings.modelCompose.column.status.fieldName,
+            //     minWidth: 50,
+            //     maxWidth: 100,
+            //     isResizable: true,
+            //     onColumnClick: this.handleColumnClick,
+            //     onRender: (model: IModel) => <span>{model.status}</span>
+            // },


remove

________________________________

In src/react/components/pages/modelCompose/modelCompose.tsx<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FOCR-Form-Tools%2Fpull%2F953%23discussion_r676724125&data=04%7C01%7Cv-yuhangzhao%40microsoft.com%7C50eacb621166458e197408d9504d4a0b%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637629115171023952%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=y5s7n37fMjVVzRPGkEQ%2BYptqckKyMs%2B%2B4kd3Z85XoSQ%3D&reserved=0>:

> +            // {
+            //     key: "column6",
+            //     name: strings.modelCompose.column.lastUpdated.headerName,
+            //     fieldName: strings.modelCompose.column.lastUpdated.fieldName,
+            //     minWidth: 175,
+            //     maxWidth: 175,
+            //     isResizable: true,
+            //     onColumnClick: this.handleColumnClick,
+            //     onRender: (model: IModel) => <span>{new Date(model.lastUpdatedDateTime).toLocaleString()}</span>,
+            // },


remove

―

File: src/common/utils.ts (hunk 1 above)
Comment (buddhawang) [reply src/common/utils.ts]: got it, thanks!

**Review by buddhawang — APPROVED** [review]

**Review by buddhawang — APPROVED** [review]

**Pull request merged**